test/hybig/HyBIG_Regression.ipynb

```diff
@@ -0,0 +1,467 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cc7d75c5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if environment_information is not None:\n",
+    "    aster_request = Request(collection=environment_information['aster_collection'],\n",
+    "                           granule_id=environment_information['aster_granules'][0],\n",
+    "                            format='image/png')                       \n",
+    "    aster_job_id = harmony_client.submit(aster_request)\n",
+    "    harmony_client.wait_for_processing(aster_job_id, show_progress=True)\n",
+    "\n",
+    "\n",
+    "    expected_files = ['ASTGTMV003_N00E022_dem.png',\n",
+    "                      'ASTGTMV003_N00E022_dem.pgw',\n",
+    "                      'ASTGTMV003_N00E022_dem.png.aux.xml']\n",
+    "\n",
+    "\n",
+    "    with TemporaryDirectory() as temp_dir:\n",
+    "        downloaded_grid_outputs = [\n",
+    "            file_future.result()\n",
+    "            for file_future\n",
+    "            in harmony_client.download_all(aster_job_id, overwrite=True, directory=temp_dir)\n",
+    "        ]\n",
+    "        \n",
+    "        \n",
+    "        reference_dir = Path('./regression_data')\n",
+    "\n",
+    "        test_files = list(map(lambda fn: Path(temp_dir) / fn, expected_files))\n",
+    "        reference_files = list(map(lambda fn: reference_dir / fn, expected_files))\n",
+    "\n",
+    "        for file_name in test_files:\n",
+    "            assert file_name.exists() == True, f'File does not exist {file_name.resolve()}'\n",
+    "        print_success('all test files generated')\n",
+    "\n",
+    "        expected_output_metadata = {'driver': 'PNG',\n",
+    "                                    'dtype': 'uint8',\n",
+    "                                    'nodata': None,\n",
+    "                                    'width': 3601,\n",
+    "                                    'height': 3601,\n",
+    "                                    'count': 4,\n",
+    "                                    'crs': CRS.from_epsg(4326),\n",
+    "                                    'transform': Affine(0.000277777777777778, 0.0, 21.9998611111111,\n",
+    "                                                        0.0, -0.000277777777777778, 1.00013888888889)}\n",
+    "\n",
+    "        test_dataset = rasterio.open(test_files[0])\n",
+    "        assert test_dataset.meta == expected_output_metadata, 'output PNG has incorrect metadata'\n",
+    "        print_success('Generated image has correct metadata.')\n",
+    "\n",
+    "        reference_dataset = rasterio.open(reference_files[0])\n",
+    "\n",
+    "        ref_image = reference_dataset.read()\n",
+    "        test_image = test_dataset.read()\n",
+    "        assert_array_almost_equal(ref_image, test_image)\n",
```

Comment:
Possibly a nit: Each test cell has quite a bit of repeated content that maybe could/should be abstracted away into some nicely named utility functions. There's a bit of a limit, as the dictionary for the expected metadata has to live in each test cell, but things like comparing the actual output files to references probably could live outside the notebook to make each test a little less busy.

Reply:
[62914dc](https://github.com/nasa/harmony-regression-tests/pull/63/commits/62914dc5baa3365b13458b93da136cc2e04601be) and [d47769e](https://github.com/nasa/harmony-regression-tests/pull/63/commits/d47769eade41d474d4dfa83b3db56c27cd0157de)